### PR TITLE
[SandboxIR] Implement Instruction flags

### PR DESCRIPTION
**Pull request by vporpo**
This patch implements sandboxir::Instruction flags.

### CODE REVIEW

**Comment by tschuett** [comment]
We ran into a similar issue: https://github.com/llvm/llvm-project/pull/93850.
There is a new NoUSWrap for geps.

**Comment by vporpo** [comment]
> There is a new NoUSWrap for geps.
These get accessed through member functions of the `GetElementPtrInst` and I think we already support them.

File: llvm/include/llvm/SandboxIR/SandboxIR.h [hunk 1]
```diff
@@ -682,6 +682,106 @@ class Instruction : public sandboxir::User {
   /// For isa/dyn_cast.
   static bool classof(const sandboxir::Value *From);
 
+  /// Determine whether the no signed wrap flag is set.
+  bool hasNoUnsignedWrap() const LLVM_READONLY {
```
Comment (aeubanks) [inline llvm/include/llvm/SandboxIR/SandboxIR.h:686]: I wouldn't mark these with LLVM_READONLY, who knows what the implementation might do under the hood

**Review by aeubanks — APPROVED** [review]
one comment, otherwise lgtm

**Review by aeubanks — APPROVED** [review]
not a fan of the LLVM-side of this API (methods should really only be on FPMathOperator), but shrug

**Comment by aeubanks** [comment]
actually, I think the `Instruction` version of some of these methods are not widely used, only the `FPMathOperator` versions. we might be able to do some cleanup and remove these from `Instruction`

**Comment by vporpo** [comment]
I am having some issue with the GenericSetter on the windows build. I will try to use a different way to get the class type from the member function pointer.

**Comment by vporpo** [comment]
> actually, I think the Instruction version of some of these methods are not widely used, only the FPMathOperator versions. we might be able to do some cleanup and remove these from Instruction

So how should we proceed? Shall I stash this until we do the cleanup?

**Comment by aeubanks** [comment]
> > actually, I think the Instruction version of some of these methods are not widely used, only the FPMathOperator versions. we might be able to do some cleanup and remove these from Instruction
> 
> So how should we proceed? Shall I stash this until we do the cleanup?

hmm it's kind of annoying to untangle given the LLVM class hierarchy. go ahead with this change, I'm giving up on refactoring this for now

**Pull request merged**